specs/layouts_manager_spec.js

```diff
@@ -31,4 +45,15 @@ describe('Layout Manager', function() {
     expect(customLayout).to.be.a('function');
     expect(customLayout('img', 'logo.png')).to.equal('imglogo.png');
   });
+
+  it('should support keeping structure in custom layouts as functions', function() {
+    var customLayout = layoutsManager.getLayout(function(type, pkg, sourceDir) {
+      return path.join(type, pkg, sourceDir);
+    });
+
+    expect(customLayout).to.be.a('function');
+    expect(customLayout('styles', 'bootstrap', 'less')).to.equal('styles/bootstrap/less');
+  });
+
+
```

Comment:
Thank you for adding all these unit tests! This is something I'd love to see more of in this project so that we can know when a feature is broken. Would you be interested in adding tests to some of the other functions as well?